conductor-client/src/main/kotlin/com/openlattice/edm/properties/PostgresTypeManager.kt

```diff
@@ -0,0 +1,133 @@
+import com.openlattice.edm.type.AssociationType
+import com.openlattice.edm.type.EntityType
+import com.openlattice.edm.type.PropertyType
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.postgres.PostgresColumn
+import com.openlattice.postgres.PostgresQuery
+import com.openlattice.postgres.PostgresTable
+import com.openlattice.postgres.ResultSetAdapters
+import com.openlattice.postgres.mapstores.AssociationTypeMapstore
+import com.openlattice.postgres.mapstores.EntityTypeMapstore
+import com.openlattice.postgres.mapstores.PropertyTypeMapstore
+import com.zaxxer.hikari.HikariDataSource
+import org.apache.olingo.commons.api.edm.FullQualifiedName
+import org.slf4j.LoggerFactory
+import java.sql.SQLException
+import java.util.Queue
+import java.util.UUID
+import java.util.function.Consumer
+import java.util.stream.Stream
+import javax.annotation.Nonnull
+
+/**
+ * @author Drew Bailey (drew@openlattice.com)
+ */
+class PostgresTypeManager(
+        val hds: HikariDataSource,
+        hazelcastInstance: HazelcastInstance
+): SchemaQueryService {
+    private val schemas = HazelcastMap.SCHEMAS.getMap(hazelcastInstance)
+    private val entityTypes = HazelcastMap.ENTITY_TYPES.getMap(hazelcastInstance)
+    private val propertyTypes = HazelcastMap.PROPERTY_TYPES.getMap(hazelcastInstance)
+    private val associationTypes = HazelcastMap.ASSOCIATION_TYPES.getMap(hazelcastInstance)
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(PostgresTypeManager::class.java)
+        private val getEnumTypeIds = "SELECT ${PostgresColumn.ID.name} FROM ${PostgresTable.ENUM_TYPES.name}"
+    }
+
+    fun getPropertyTypesInNamespace(namespace: String?): Iterable<PropertyType> {
+        return propertyTypes.values(Predicates.equal(PropertyTypeMapstore.NAMESPACE_INDEX, namespace))
+    }
+
+    fun getPropertyTypes(): Iterable<PropertyType> {
+        return propertyTypes.values
+    }
+
+    fun getIdsForQuery(query: String): Iterable<UUID> {
+        try {
+            hds.connection.use { connection ->
+                connection.prepareStatement(query).use { ps ->
+                    val result: MutableList<UUID> = Lists.newArrayList()
+                    ps.executeQuery().use { rs ->
+                        while (rs.next()) {
+                            result.add(ResultSetAdapters.id(rs))
+                        }
+                        connection.close()
+                        return result
+                    }
+                }
+            }
+        } catch (e: SQLException) {
+            logger.debug("Unable to load ids for query: {}", query, e)
+            return ImmutableList.of()
+        }
+    }
+
+    fun getEntityTypes(): Iterable<EntityType> {
+        return entityTypes.values
+    }
+
+    fun getEntityTypesStrict(): Iterable<EntityType> {
+        return entityTypes.values(Predicates.equal<UUID, EntityType>(EntityTypeMapstore.CATEGORY_INDEX, SecurableObjectType.EntityType.name))
```

Comment:
I don't think you need the `.name` here -- it should match on the enum value itself (not sure if .name will work actually)